File: 2025-02-15-issue-csrf-token-between-spring-security-and-react/action-in-blog/backend/src/main/java/action/in/blog/controller/TodoController.java
package action.in.blog.controller;

import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

record Todo(
        String content
) {
}

@RestController
public class TodoController {

    @PostMapping("/api/todos")
    void createTodos(@RequestBody Todo todo) {
        System.out.println(todo);
    }
}

File: 2024-07-16-how-to-deploy-react-application-on-jar-package/action-in-blog/backend-gradle/src/main/kotlin/action/in/blog/cotroller/TodoController.kt
package action.`in`.blog.backend.cotroller

import action.`in`.blog.backend.domain.Todo
import action.`in`.blog.backend.repository.TodoRepository
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/todos")
class TodoController(
    private val todoRepository: TodoRepository
) {

    @GetMapping
    fun getTodos(): MutableList<Todo> = todoRepository.findAll()
        .map { Todo.from(it) }
        .toMutableList()

    @PostMapping
    fun createTodo(@RequestBody todo: Todo) = todoRepository.save(todo.toEntity())
}